src/libraries/Microsoft.Extensions.Configuration.Binder/gen/ConfigurationBindingGenerator.Parser.cs

```diff
@@ -65,6 +65,15 @@ public Parser(SourceProductionContext context, KnownTypeSymbols typeSymbols, Imm
                     }
                 }
 
+                foreach (var typeSymbol in _createdSpecs.Keys)
+                {
+                    if (IsEnum(typeSymbol))
+                    {
+                        _sourceGenSpec.EmitEnumParseMethod = true;
+                        break;
+                    }
+                }
+
```

Comment:
It would be better to set the property here instead - https://github.com/dotnet/runtime/blob/3fac334c3ab3a61e41a2913ad549e666e91bdeed/src/libraries/Microsoft.Extensions.Configuration.Binder/gen/ConfigurationBindingGenerator.Parser.cs#L239.